### ISSUE TITLE
How to get only score?

### ISSUE THREAD

**Issue by devender-yadav** [PyCQA/pylint#3024]
## Question

Is there any way to disable output and show only score via command line or programmatically?

**Comment by PCManticore** [comment]
No, the score is not the purpose of pylint, the output is.

**Comment by devender-yadav** [comment]
@PCManticore what's wrong with "only" score?

It can be used in CI-CD pipelines to ensue code below a certain threshold value doesn't get deployed.

**Comment by techspeque** [comment]
@devender-yadav There is a way to display only the score: `pipenv run pylint -E $(find ./src/ -name "*.py") --score=y --reports=n` the `reports=n` suppresses other output and `--score=y` ensures that the score is being displayed. You can then parse it in a way you wish.

**Comment by hippo91** [comment]
@techspeque thanks for taking time to answer to this old post. :+1:

**Comment by kasiagauza** [comment]
@techspeque It doesn't work. `--reports=n` and `--score=y` show exactly the same output as w/o the flags.

**Comment by Pierre-Sassoulas** [comment]
It's not possible, as PCManticore said, the messages are the point. You have option to make pylint fail under a score threshold, a 0 to 32 exit code to fine tune what type of error you accept that are more conveniant than parsing the output in CI

**Comment by kasiagauza** [comment]
My case is that I would like to save the detailed report to a JSON file and display only the summary in a command line, e.g. `pylint --output-format=parseable:<some_file>,summary <directory>` -- note the summary value.

**Comment by Pierre-Sassoulas** [comment]
See #4741

**Comment by lettow-humain** [comment]
Cross posting this from #4741 because these two issues come up when you search for the problem of only getting the score.

For everyone (like me) ending up here in search for a one line solution to use in CI/CD:

Shell:
```bash
pylint -f json2 yourSickProgram.py | python3 -c "import sys, json; print(json.load(sys.stdin)['statistics']['score'])"
```

For everyone looking to only get the score programmatically without all the messages:

Python:
```python
"""
Pipe standard output stdout to /dev/null temporarily
to avoid excessive output
"""

import os
import sys
import pylint.lint

# Save original pipeline
ogPipe = sys.stdout 
# Create temporary pipeline to /dev/null
tmpPipe = open(os.devnull, 'w')
# Pipe stdout to temporary pipeline
sys.stdout = tmpPipe

# Run pylint on your file(s) and save score in variable
result = round(pylint.lint.Run(
    ['yourSickProgram.py'], exit=False).linter.stats.global_note, 2)

# Set stdout pipe back to original
sys.stdout = ogPipe
# Close temporary pipeline
tmpPipe.close()
# Print result
print(result)
```
